ShenzhenMod/ShenzhenLocator.cs

```diff
@@ -119,11 +124,28 @@ private IEnumerable<string> GetMacSearchPaths()
 
         public static string FindUnpatchedShenzhenExecutable(string shenzhenDir)
         {
+            sm_log.InfoFormat("Fetching the latest list of unpatched hashes");
+
+            WebRequest webRequest = WebRequest.Create("https://raw.githubusercontent.com/gtw123/ShenzhenMod/master/hashes.json");
+            WebResponse response = webRequest.GetResponse();
+
+            if (((HttpWebResponse)response).StatusDescription != "OK")
```

Comment:
From what I can tell, GetResponse already throws an exception if it has an error, so it's probably not necessary to check the status here. But the exception doesn't give much context, so it might be better to wrap GetResponse in a try/catch and throw a new exception with some context.

Reply:
I now have it throw a clear and generic "Error occurred fetching hash list", followed by the specific network error.